### DISCUSSION TITLE
how to reload excel ribbon dynamic dropdown boxes

### DISCUSSION THREAD

**Discussion by michael-hgl** [fernandreu/office-ribbonx-editor#245] [Q&A]
The VBA ribbon callback functions load the dynamic dropdown boxes when
you first select your Tab on the Excel ribbon.

But I need to know how to later clear and reload these dropdown boxes with
different info.

Anybody know how to do this or point me to some info?

Thanks,
Michael

**Comment by thefasquelle** [comment]
@michael-hgl This is not the proper place to ask this. This repo is only about Office RibbonX Editor, not VBA programming. You should ask this on stackoverflow or a dedicated forum (mrexcel.com, excelforum.com ...). I will answer your question but in the future please don't put off-topic question here.

You need to use the OnLoad callback to store your ribbon object. Then, at any time in your VBA code you can call Ribbon.InvalidateControl(dropDownId) to invalidate just your drop down menu, or Ribbon.InvalidateRibbon() to invalidate the entire ribbon. The next time Excel needs to display an invalidated control, it will call its callback again.

You may search for community posts about these topics and read the official documentation about ribbon customization
- https://learn.microsoft.com/en-us/office/vba/api/office.iribbonui#example
- https://learn.microsoft.com/en-us/office/vba/api/office.iribbonui.invalidatecontrol
- https://learn.microsoft.com/en-us/previous-versions/office/developer/office-2007/aa338202(v=office.12)
- https://learn.microsoft.com/en-us/openspecs/office_standards/ms-customui/31f152d6-2a5d-4b50-a867-9dbc6d01aa43

**Comment by michael-hgl** [comment]
Okay, thanks for the info, and sorry about posting in wrong forum.

Michael
